File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/wallsentinels/domain/WallSentinelsObject.java
package com.zwstudio.logicpuzzlesandroid.puzzles.wallsentinels.domain;


public abstract class WallSentinelsObject {
    public String objAsString() {
        return "empty";
    }
    public static WallSentinelsObject objFromString(String str) {
        switch (str) {
            case "marker":
                return new WallSentinelsMarkerObject();
            case "wall":
                return new WallSentinelsWallObject();
            default:
                return new WallSentinelsEmptyObject();
        }
    }
}

File: app/src/main/java/com/zwstudio/logicpuzzlesandroid/puzzles/wallsentinels/WallSentinelsObject.kt
package com.zwstudio.logicpuzzlesandroid.puzzles.wallsentinels

import com.zwstudio.logicpuzzlesandroid.common.domain.Position

enum class WallSentinelsObject {
    Empty, Marker, HintWall, HintLand, Wall
}

class WallSentinelsGameMove(val p: Position, var obj: WallSentinelsObject = WallSentinelsObject.Empty)
